This patch was adapted from a 2.6 patch which came from upstream:
http://bugs.python.org/issue19561

--- Python-3.5.0a4/Include/pyport.h.~1~	2015-04-20 00:37:51.000000000 -0700
+++ Python-3.5.0a4/Include/pyport.h	2015-04-20 14:17:45.716965529 -0700
@@ -640,11 +640,6 @@
 in platform-specific #ifdefs.
 **************************************************************************/
 
-#ifdef SOLARIS
-/* Unchecked */
-extern int gethostname(char *, int);
-#endif
-
 #ifdef HAVE__GETPTY
 #include <sys/types.h>          /* we need to import mode_t */
 extern char * _getpty(int *, int, mode_t, int);
